src/Rule/Nested.php

```diff
@@ -132,18 +135,13 @@ private function prepareRules(iterable|object|string|null $source): ?iterable
 
         if ($source instanceof RulesProviderInterface) {
             $rules = $source->getRules();
-            return $this->normalizeRules ? $this->normalizeRules($rules) : $rules;
+        } elseif (!$source instanceof Traversable && !is_array($source)) {
+            $rules = (new AttributesRulesProvider($source, $this->rulesPropertyVisibility))->getRules();
```

Comment:
Why we don't can make return here?

Reply:
https://github.com/yiisoft/validator/pull/303#discussion_r965789351